### DISCUSSION TITLE
corpora.train.augmenter throws extra fields not permitted error

### DISCUSSION THREAD

**Discussion by scellat** [explosion/spaCy#6519] [Help: Coding & Implementations] [answered]
I am running the augmentation feature of the spacy 3 and I followed the instructions provided in the [documentation.](https://nightly.spacy.io/usage/training) I am adding the following default augmenter to my working code

 ```
[corpora.train.augmenter]
@augmenters = "spacy.orth_variants.v1"
# Percentage of texts that will be augmented / lowercased
level = 0.1
lower = 0.5

[corpora.train.augmenter.orth_variants]
@readers = "srsly.read_json.v1"
path = "corpus/orth_variants.json"
```

but it throws the following error

```
✘ Config validation error
corpora.train -> augmenter   extra fields not permitted
```
Any idea on what might be causing this error?

I am using spaCy 3.0.0rc2
* Operating System: Mac
* Python Version Used: 3.7.9 
* spaCy Version Used:3.0.0rc2  
* Environment Information:

**Comment by svlandeg** [comment]
Can you paste the full definiton of your `corpora.train` block? Are you using `spacy.Corpus.v1`?

**Comment by scellat** [comment]
I am not using `spacy.Corpus.V1`

````
[paths]
data_cfg = data/dataset.yaml
train = ${paths.data_cfg}
dev = ${paths.data_cfg}
raw = null
init_tok2vec = null
vectors = null

[system]
seed = 0
gpu_allocator = null

[corpora]

[corpora.train]
@readers = "YReader.V1"
config_path = ${paths:train}
mode = train

[corpora.train.augmenter]
@augmenters = "spacy.orth_variants.v1"
# Percentage of texts that will be augmented / lowercased
level = 0.1
lower = 0.5

[corpora.train.augmenter.orth_variants]
@readers = "srsly.read_json.v1"
path = "corpus/orth_variants.json"
```

**Comment by svlandeg** [comment, marked as answer]
Ah, that explains it!

The way the config works, is that you're hierarchically defining functions. Blocks lower down in the hierarchy are fed as parameters into the upper function. So, `config_path` is fed as parameter into your `YReader.V1` reader. But also `augmenter` is fed into this same `YReader.V1`. So if you want to provide data augmentation also in your custom reader, you'll have to implement it where you implemented that specific reader.

For inspiration, you can consult the code from `spacy.Corpus.v1` [here](https://github.com/explosion/spaCy/blob/develop/spacy/training/corpus.py). You can find the definitions of the existing augmenters [here](https://github.com/explosion/spaCy/blob/develop/spacy/training/augment.py). Their signature is `Callable[["Language", Example], Iterator[Example]]` which means that you can call them with an `nlp` and an `example`, and get an iterator of (slightly modified) examples back. So you should be able to incorporate that functionality into your custom reader.

**Comment by scellat** [comment]
That info helps a lot, thank you!

**Comment by shresthadipak** [comment]
Problem Solved:
textcat -> architecture extra fields not permitted in spaCy Error got Solved. This problem will be solved by downgrading the version of spaCY. Run this code to downgrade the spaCy Version to 2.3.6

`pip install -U spacy==2.3.6`

This is work for me. Hope this will also help you.

**Reply by svlandeg** [reply]
I'm not sure what you mean by this post. It looks like you've encountered a different error as the one originally discussed in this topic, but moreover downgrading to 2.3.6 pulls in a much older version of spaCy that didn't even work with the training config files yet?

**Reply by shresthadipak** [reply]
I got the error of  "textcat -> architectureextra fields not permitted" but I could solve this error by downgrading the version of spaCy to 2.3.6. This worked well for me. For further process of Training a Text Categorizer Model and Make predictions have no problem occur.
